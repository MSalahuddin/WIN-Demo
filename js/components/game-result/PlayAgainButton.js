import React from 'react';
import styled from 'styled-components/native';
import PropTypes from 'prop-types';

import Text, { SIZE, FONT_FAMILY } from '../common/Text';
import GameCardButton from '../game-room/GameCardButton';
import { gameResultScreenStrings } from '../../stringConstants';
import { scale, scaleHeight, heightRatio } from '../../platformUtils';
import { color } from '../../styles';
import SimpleButton from '../common/SimpleButton';
import { coin } from '../../../assets/images';

const TextContainer = styled.View`
  flex: ${({areTokenLow})=> !areTokenLow ? 1 : 0.5};;
  background-color: ${({areTokenLow})=>!areTokenLow?color.white:color.transparent};
  border-radius: 8;
  justify-content: flex-end;
  align-items: center;
  padding-bottom: ${heightRatio > 1 ? scaleHeight(10) : scaleHeight(3)};
`;
const ButtonContentContainer = styled.View`
flex-direction : row
`
const CoinIcon = styled.Image`
  margin-horizontal: ${({ marginHorizontal }) => scale(marginHorizontal)};
  height: ${({ size }) => scale(size)};
  width: ${({ size }) => scale(size)};
`;
const ButtonWrapper = styled(SimpleButton)`
`;
const GameButtonContainer = styled.View`
  align-items:center;
  position: absolute;
  top: -${scaleHeight(50)};
  left: ${scale(10)};
  right: ${scale(10)};
`;

const PlayAgainButton = ({ label, tokensCost, isFree, onPress, areTokenLow, tokenAmount , localizedPrice ,...rest }) => {
  return (
    <>
      <TextContainer areTokenLow={areTokenLow}>
        {!areTokenLow && <Text allowFontScaling={false} fontFamily={FONT_FAMILY.REGULAR} size={SIZE.XSMALL}>
          {label}
        </Text>}
      </TextContainer>
      <GameButtonContainer>
        {!areTokenLow ? <GameCardButton
          isFree={isFree}
          isLarge
          playLabel={gameResultScreenStrings.playAgain}
          tokensCost={tokensCost}
          onPress={onPress}
          testID="play-again-button"
          {...rest}
        />:
        <ButtonWrapper
        borderRadius={scale(50)}
        height={scale(80)}
        width={271}
        onPress={onPress}
        >
          <ButtonContentContainer>
            <Text 
              size={SIZE.LARGE} 
              color={color.darkNavyBlue} 
              fontFamily={FONT_FAMILY.BOLD} 
              marginTop={Platform.OS==="android"? 0 : 5}>{localizedPrice}</Text>
              <CoinIcon source={coin} size={25} marginHorizontal={8} resizeMode="contain" />
            <Text
            size={SIZE.LARGE} 
            color={color.darkNavyBlue} 
            fontFamily={FONT_FAMILY.BOLD} 
            marginTop={Platform.OS==="android"? 0 : 5}>{tokenAmount}</Text>
          </ButtonContentContainer>
        </ButtonWrapper>
        }
      </GameButtonContainer>
    </>
  );
};

PlayAgainButton.defaultProps = {
  isFree: false,
  label: gameResultScreenStrings.prizeWaitingInVault
};

PlayAgainButton.propTypes = {
  tokensCost: PropTypes.number.isRequired,
  isFree: PropTypes.bool,
  label: PropTypes.string,
  localizedPrice:PropTypes.number,
  tokenAmount:PropTypes.number,
  onPress: PropTypes.func.isRequired
};

export default PlayAgainButton;
