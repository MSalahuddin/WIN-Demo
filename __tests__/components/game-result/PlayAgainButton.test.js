import React from 'react';
import { render, fireEvent } from 'react-native-testing-library';

import PlayAgainButton from '../../../js/components/game-result/PlayAgainButton';
import MockProvider from '../../MockProvider';

describe('Component PlayAgainButton', () => {
  const onPress = jest.fn();
  test('renders correctly', () => {
    const output = render(
      <MockProvider>
        <PlayAgainButton tokensCost={1} onPress={onPress} />
      </MockProvider>
    );
    expect(output).toMatchSnapshot();
  });

  test('renders correctly when isFree', () => {
    const output = render(
      <MockProvider>
        <PlayAgainButton tokensCost={1} onPress={onPress} isFree />
      </MockProvider>
    );
    expect(output).toMatchSnapshot();
  });

  test('calls onPress when play again button is pressed', () => {
    const { getByTestId } = render(
      <MockProvider>
        <PlayAgainButton tokensCost={1} onPress={onPress} isFree />
      </MockProvider>
    );
    fireEvent(getByTestId('play-again-button'), 'press');
    expect(onPress).toHaveBeenCalled();
  });
});
