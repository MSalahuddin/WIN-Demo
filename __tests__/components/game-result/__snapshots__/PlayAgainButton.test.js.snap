// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component PlayAgainButton renders correctly 1`] = `
Array [
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "borderRadius": 8,
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "justifyContent": "flex-end",
          "paddingBottom": 20,
        },
      ]
    }
  >
    <Text
      alignCenter={false}
      allowFontScaling={false}
      color="#000000"
      fontFamily="REGULAR"
      isUnderlined={false}
      isUppercase={false}
      maxFontSizeMultiplier={1.3}
      size="XSMALL"
      style={
        Array [
          Object {
            "color": "#000000",
            "fontFamily": "CalibreRegular",
            "fontSize": 24,
            "justifyContent": "center",
            "textAlign": "left",
            "textDecorationColor": "#000000",
            "textDecorationLine": "none",
            "textDecorationStyle": "solid",
            "textTransform": "none",
          },
        ]
      }
    >
      Your prize will be waiting in your prize vault!
    </Text>
  </View>,
  <View
    style={
      Array [
        Object {
          "left": 15,
          "position": "absolute",
          "right": 15,
          "top": -100,
        },
      ]
    }
  >
    <View
      onLayout={[Function]}
      style={
        Array [
          Object {
            "alignItems": "center",
            "justifyContent": "center",
            "marginHorizontal": 10,
            "marginVertical": 10,
          },
        ]
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "rgba(96,202,142,1)",
            "borderRadius": 80,
            "height": 160,
            "justifyContent": "center",
            "opacity": 1,
            "width": 542,
          }
        }
        testID="play-again-button"
      >
        <View
          borderRadius={40}
          colorScheme="rgba(128,229,173,1)"
          height={80}
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "rgba(128,229,173,1)",
                "borderRadius": 80,
                "height": 150,
                "justifyContent": "center",
                "width": 536,
              },
            ]
          }
          width={271}
        >
          <View
            borderRadius={40}
            colorScheme="rgba(114,213,158,1)"
            height={80}
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "rgba(114,213,158,1)",
                  "borderRadius": 100,
                  "height": 146,
                  "justifyContent": "center",
                  "marginTop": 8,
                  "width": 536,
                },
              ]
            }
            width={271}
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "justifyContent": "center",
                  },
                ]
              }
            >
              <Text
                alignCenter={false}
                allowFontScaling={true}
                color="#FFFFFF"
                fontFamily="MEDIUM"
                isUnderlined={false}
                isUppercase={false}
                marginTop={5}
                maxFontSizeMultiplier={1.3}
                size="XLARGE"
                style={
                  Array [
                    Object {
                      "color": "#FFFFFF",
                      "fontFamily": "Calibre-Medium",
                      "fontSize": 36,
                      "justifyContent": "center",
                      "textAlign": "left",
                      "textDecorationColor": "#FFFFFF",
                      "textDecorationLine": "none",
                      "textDecorationStyle": "solid",
                      "textTransform": "none",
                    },
                    Object {
                      "marginHorizontal": 7.5,
                      "marginTop": 10,
                    },
                  ]
                }
              >
                Play again
              </Text>
              <Image
                marginHorizontal={8}
                resizeMode="contain"
                size={40}
                source={1}
                style={
                  Array [
                    Object {
                      "height": 60,
                      "marginHorizontal": 12,
                      "width": 60,
                    },
                  ]
                }
              />
              <View
                style={
                  Array [
                    Object {
                      "justifyContent": "center",
                    },
                  ]
                }
              >
                <Text
                  alignCenter={false}
                  allowFontScaling={true}
                  color="#FFFFFF"
                  fontFamily="BOLD"
                  isUnderlined={false}
                  isUppercase={false}
                  marginTop={5}
                  maxFontSizeMultiplier={1.3}
                  size="XLARGE"
                  style={
                    Array [
                      Object {
                        "color": "#FFFFFF",
                        "fontFamily": "CalibreBold",
                        "fontSize": 36,
                        "justifyContent": "center",
                        "textAlign": "left",
                        "textDecorationColor": "#FFFFFF",
                        "textDecorationLine": "none",
                        "textDecorationStyle": "solid",
                        "textTransform": "none",
                      },
                      Object {
                        "marginHorizontal": 7.5,
                        "marginTop": 10,
                      },
                    ]
                  }
                >
                  1
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>,
]
`;

exports[`Component PlayAgainButton renders correctly when isFree 1`] = `
Array [
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "borderRadius": 8,
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "justifyContent": "flex-end",
          "paddingBottom": 20,
        },
      ]
    }
  >
    <Text
      alignCenter={false}
      allowFontScaling={false}
      color="#000000"
      fontFamily="REGULAR"
      isUnderlined={false}
      isUppercase={false}
      maxFontSizeMultiplier={1.3}
      size="XSMALL"
      style={
        Array [
          Object {
            "color": "#000000",
            "fontFamily": "CalibreRegular",
            "fontSize": 24,
            "justifyContent": "center",
            "textAlign": "left",
            "textDecorationColor": "#000000",
            "textDecorationLine": "none",
            "textDecorationStyle": "solid",
            "textTransform": "none",
          },
        ]
      }
    >
      Your prize will be waiting in your prize vault!
    </Text>
  </View>,
  <View
    style={
      Array [
        Object {
          "left": 15,
          "position": "absolute",
          "right": 15,
          "top": -100,
        },
      ]
    }
  >
    <View
      onLayout={[Function]}
      style={
        Array [
          Object {
            "alignItems": "center",
            "justifyContent": "center",
            "marginHorizontal": 10,
            "marginVertical": 10,
          },
        ]
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "rgba(96,202,142,1)",
            "borderRadius": 80,
            "height": 160,
            "justifyContent": "center",
            "opacity": 1,
            "width": 542,
          }
        }
        testID="play-again-button"
      >
        <View
          borderRadius={40}
          colorScheme="rgba(128,229,173,1)"
          height={80}
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "rgba(128,229,173,1)",
                "borderRadius": 80,
                "height": 150,
                "justifyContent": "center",
                "width": 536,
              },
            ]
          }
          width={271}
        >
          <View
            borderRadius={40}
            colorScheme="rgba(114,213,158,1)"
            height={80}
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "rgba(114,213,158,1)",
                  "borderRadius": 100,
                  "height": 146,
                  "justifyContent": "center",
                  "marginTop": 8,
                  "width": 536,
                },
              ]
            }
            width={271}
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "justifyContent": "center",
                  },
                ]
              }
            >
              <Text
                alignCenter={false}
                allowFontScaling={true}
                color="#FFFFFF"
                fontFamily="MEDIUM"
                isUnderlined={false}
                isUppercase={false}
                marginTop={5}
                maxFontSizeMultiplier={1.3}
                size="XLARGE"
                style={
                  Array [
                    Object {
                      "color": "#FFFFFF",
                      "fontFamily": "Calibre-Medium",
                      "fontSize": 36,
                      "justifyContent": "center",
                      "textAlign": "left",
                      "textDecorationColor": "#FFFFFF",
                      "textDecorationLine": "none",
                      "textDecorationStyle": "solid",
                      "textTransform": "none",
                    },
                    Object {
                      "marginHorizontal": 7.5,
                      "marginTop": 10,
                    },
                  ]
                }
              >
                Play again
              </Text>
              <Image
                marginHorizontal={8}
                resizeMode="contain"
                size={40}
                source={1}
                style={
                  Array [
                    Object {
                      "height": 60,
                      "marginHorizontal": 12,
                      "width": 60,
                    },
                  ]
                }
              />
              <View
                style={
                  Array [
                    Object {
                      "justifyContent": "center",
                    },
                  ]
                }
              >
                <Text
                  alignCenter={false}
                  allowFontScaling={true}
                  color="#FFFFFF"
                  fontFamily="BOLD"
                  isUnderlined={false}
                  isUppercase={false}
                  marginTop={5}
                  maxFontSizeMultiplier={1.3}
                  size="XLARGE"
                  style={
                    Array [
                      Object {
                        "color": "#FFFFFF",
                        "fontFamily": "CalibreBold",
                        "fontSize": 36,
                        "justifyContent": "center",
                        "textAlign": "left",
                        "textDecorationColor": "#FFFFFF",
                        "textDecorationLine": "none",
                        "textDecorationStyle": "solid",
                        "textTransform": "none",
                      },
                      Object {
                        "marginHorizontal": 7.5,
                        "marginTop": 10,
                      },
                    ]
                  }
                >
                  1
                </Text>
              </View>
              <View
                style={
                  Array [
                    Object {
                      "borderBottomWidth": 3,
                      "borderColor": "#F06B59",
                      "height": 1,
                      "position": "absolute",
                      "width": "65%",
                    },
                  ]
                }
                width="65%"
              />
            </View>
            <View
              style={
                Array [
                  Object {
                    "position": "absolute",
                    "right": -7.5,
                    "top": -56,
                  },
                ]
              }
              top={-28}
            >
              <View
                onLayout={[Function]}
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "justifyContent": "center",
                      "marginHorizontal": 10,
                      "marginVertical": 10,
                    },
                  ]
                }
              >
                <View
                  accessible={true}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "rgba(197,78,62,1)",
                      "borderRadius": 106,
                      "height": 70,
                      "justifyContent": "center",
                      "opacity": 1,
                      "width": 148,
                    }
                  }
                  testID="overlay-button"
                >
                  <View
                    borderRadius={53}
                    colorScheme="rgba(255,132,115,1)"
                    height={35}
                    style={
                      Array [
                        Object {
                          "alignItems": "center",
                          "backgroundColor": "rgba(255,132,115,1)",
                          "borderRadius": 106,
                          "height": 60,
                          "justifyContent": "center",
                          "width": 142,
                        },
                      ]
                    }
                    width={74}
                  >
                    <View
                      borderRadius={53}
                      colorScheme="rgba(240, 107, 89, 1)"
                      height={35}
                      style={
                        Array [
                          Object {
                            "alignItems": "center",
                            "backgroundColor": "rgba(240, 107, 89, 1)",
                            "borderRadius": 126,
                            "height": 56,
                            "justifyContent": "center",
                            "marginTop": 8,
                            "width": 142,
                          },
                        ]
                      }
                      width={74}
                    >
                      <Text
                        alignCenter={false}
                        allowFontScaling={true}
                        color="#FFFFFF"
                        fontFamily="BOLD ITALIC"
                        isUnderlined={false}
                        isUppercase={false}
                        marginTop={5}
                        maxFontSizeMultiplier={1.3}
                        size="SMALL"
                        style={
                          Array [
                            Object {
                              "color": "#FFFFFF",
                              "fontFamily": "CalibreBoldItalic",
                              "fontSize": 27,
                              "justifyContent": "center",
                              "textAlign": "left",
                              "textDecorationColor": "#FFFFFF",
                              "textDecorationLine": "none",
                              "textDecorationStyle": "solid",
                              "textTransform": "none",
                            },
                            Object {
                              "marginHorizontal": 7.5,
                              "marginTop": 10,
                            },
                          ]
                        }
                      >
                        FREE!
                      </Text>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>,
]
`;
